### PR TITLE
Update: Hide datetime column on all timegrain

### PR DESCRIPTION
**Pull request by mattgraham1995**
## Description
We recently decided that when the request has a timegrain of `all`, we will hide the dateTime column in the table visualization.

## Proposed Changes
- Change table model validations to check for the existence of the datetime column based on the timegrain
- Change `rebuildConfig` in table model to handle adding or removing the datetime column when the timegrain changes

### CODE REVIEW

File: packages/core/addon/models/table.js [hunk 1]
```diff
@@ -54,19 +58,21 @@ export default VisualizationBase.extend(Validations, {
...
-      dateColumn,
-      ...buildDimensionColumns(dimensions, columnIndex),
-      ...buildMetricColumns(metrics, columnIndex)
-    ];
+    let newColumns = [...buildDimensionColumns(dimensions, columnIndex), ...buildMetricColumns(metrics, columnIndex)];
+
+    //Add DateTime column when on timegrain other than All
+    if (timeGrain !== 'all') {
+      newColumns.unshift(dateColumn);
+    }
```
Comment (jkusa) [inline packages/core/addon/models/table.js]: Have you seen this technique?
```js
const newColumns = [
...(timeGrain !== 'all') && [dateColumn], 
...buildDimensionColumns(dimensions, columnIndex),
...buildMetricColumns(metrics, columnIndex)
];
```

Comment (mattgraham1995) [reply packages/core/addon/models/table.js]: I haven't. I didn't know you could do that with the spread operator

**Comment by jkusa** [comment]
@mattgraham1995 as mentioned offline, can you double check this won't have a heavy impact to column customization for reports with `all` time grain. 

Also, please check dashboards, as they don't verify the visualization model.

File: packages/core/addon/models/table.js (hunk 1 above)
Comment (mattgraham1995) [reply packages/core/addon/models/table.js]: Screen Shot 2019-04-01 at 3 40 42 PM

Couldn't get the spread to work like that so I changed it a different way to not use `unshift`

**Comment by mattgraham1995** [comment]
@jkusa Checked dashboards and it works as expected. Checked existing reports with customized tables and they all retain their customizations while dropping the date column.

**Pull request merged**